x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/plugin/IndexResolutionIT.java

```diff
@@ -86,6 +86,9 @@ public void testResolvesDataStream() {
         try (var response = run(syncEsqlQueryRequest("FROM data-stream-1"))) {
             assertOk(response);
         }
+        try (var response = run(syncEsqlQueryRequest("FROM data-stream-1::data"))) {
```

Comment:
Can this be applied to all kinds of indices? 
In case, maybe we can randomize some of our existing tests (conditionally adding `::data`), that already validate the query output.